source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -858,16 +882,33 @@ bool JsonTranscoderFilter::readToBuffer(Protobuf::io::ZeroCopyInputStream& strea
   return false;
 }
 
+void JsonTranscoderFilter::onDestroy() {
+  if (request_data_.length()) {
+    stats_->transcoder_request_buffer_bytes_.sub(request_data_.length());
+  }
+  if (response_data_.length()) {
+    stats_->transcoder_response_buffer_bytes_.sub(response_data_.length());
+  }
+  if (request_in_.bytesStored()) {
+    stats_->transcoder_request_buffer_bytes_.sub(request_in_.bytesStored());
+  }
+  if (response_in_.bytesStored()) {
+    stats_->transcoder_response_buffer_bytes_.sub(response_in_.bytesStored());
+  }
```

Comment:
here again I think we can collect up in a local the total change we are making to the gauge, and `sub` it once.

Reply:
Done.